platform/ios/jazzy.yml

```diff
@@ -23,53 +26,61 @@ custom_categories:
       - MGLMapViewDelegate
       - MGLStyle
       - MGLUserTrackingMode
-  - name: Annotations
+  - name: User Interaction
     children:
-      - MGLAnnotation
-      - MGLAnnotationImage
-      - MGLAnnotationView
       - MGLCalloutView
       - MGLCalloutViewDelegate
+  - name: Protocols and Abstract Classes
+    children:
+      - MGLAnnotation
+      - MGLOverlay
+      - MGLShape
       - MGLMultiPoint
-      - MGLMultiPolygon
-      - MGLMultiPolyline
+      - MGLFeature
+  - name: Primitive Shapes
+    children:
       - MGLPointAnnotation
       - MGLPointCollection
       - MGLPolygon
       - MGLPolyline
-      - MGLOverlay
-      - MGLShape
+      - MGLMultiPolygon
+      - MGLMultiPolyline
       - MGLShapeCollection
+  - name: Annotations
+    children:
+      - MGLAnnotationImage
+      - MGLAnnotationView
       - MGLUserLocation
       - MGLUserLocationAnnotationView
-  - name: Map Data
+  - name: Style Features
     children:
-      - MGLFeature
-      - MGLMultiPolygonFeature
-      - MGLMultiPolylineFeature
-      - MGLPointCollectionFeature      
       - MGLPointFeature
       - MGLPolygonFeature
       - MGLPolylineFeature
+      - MGLMultiPolygonFeature
+      - MGLMultiPolylineFeature
+      - MGLPointCollectionFeature      
       - MGLShapeCollectionFeature
+  - name: Style Sources
+    children:
+      - MGLSource
+      - MGLTileSet
+      - MGLGeoJSONSource
+      - MGLRasterSource
+      - MGLVectorSource
+  - name: Style Layers (Abstract)
```

Comment:
I wonder whether dividing classes by abstractness is really a good idea for the style layer classes, which were designed to be a nested inheritance tree, with each level sort of peeling a layer off the onion. On the one hand, categorizing them this way helps developers find the classes they can safely initialize. On the other hand, it also obscures the relationships between each concrete class and the abstract superclasses that contribute to its properties and initializers.

To be sure, one alphabetical list wasn’t especially enlightening, either, but what about a flattened inheritance hierarchy like this:

* MGLStyleLayer
* MGLBackgroundStyleLayer
* MGLForegroundStyleLayer
* MGLRasterStyleLayer
* MGLVectorStyleLayer
* MGLCircleStyleLayer
* MGLFillStyleLayer
* _MGLFillExtrusionStyleLayer would go here_
* MGLLineStyleLayer
* MGLSymbolStyleLayer